app/helpers/automated_tests_client_helper.rb

```diff
@@ -1,6 +1,4 @@
-require 'net/ssh'
-require 'net/scp'
-require 'json'
+require File.join(Rails.root, 'lib', 'automated_tests', 'server', 'automated_tests_server')

```

Comment:
I struggled with this for a bit, let me try to recap the critical points:
1) We want a developer instance to work out of the box locally, without having to install the server separately.
2) A production instance with ate_server_host == 'localhost' is perfectly viable, it's just not our production instance.
3) Rails 5 will want to eagerly load all the code to detect potential concurrency problems, i.e. all the relevant code needs to be loaded at boot time.

What you are suggesting is basically a dynamic require, right? That won't be possible/supported with Rails5 as far as I understand (http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#autoloading-is-disabled-after-booting-in-the-production-environment), I can't figure out a way to follow Rails5's philosophy and drop the file in some production environments (note the 'some', it's not in all, which would be somewhat easier). This made me think that after all the server is still dependent on Markus for the communication and not an entirely separate project with a specified communication protocol, so I went for the compromise where the con is having the server ruby file loaded by clients even if they don't need it.